Volume/Views/MainView/MagazineReaderView.swift

```diff
@@ -0,0 +1,222 @@
+//
+//  MagazineDetail.swift
+//  Volume
+//
+//  Created by Justin Ngai on 9/3/2022.
+//  Copyright © 2022 Cornell AppDev. All rights reserved.
+//
+
+import SwiftUI
+import PDFKit
+import AppDevAnalytics
+import Combine
+import LinkPresentation
+import SDWebImageSwiftUI
+
+struct MagazineReaderView: View {
+    @Environment(\.presentationMode) var presentationMode: Binding<PresentationMode>
+    @EnvironmentObject private var userData: UserData
+    
+    let magazine: Magazine
+    let navigationSource: NavigationSource
+    @State private var bookmarkRequestInProgress = false
+    @State private var cancellableShoutoutMutation: AnyCancellable?
+    @State private var showToolbars = true
+    
+    private var isShoutoutsButtonEnabled: Bool {
+        return userData.canIncrementMagazineShoutouts(magazine)
+    }
+    
+    private var navbar: some View {
+            ZStack {
+                Rectangle()
+                    .foregroundColor(.white)
+                    .shadow(color: .black.opacity(0.2), radius: 2, y: 3)
```

Comment:
factor these constants out into a `Constants` struct

Reply:
I think it would get too confusing due to how many constants there are. I feel like it would make more sense to future developers to change these constants exactly where they are used instead of trying to match variable names. But lmk if you still think they should be pulled out and I can do it

Reply:
Yup I see your point. However let's say you decide to use this constant in more than one place. Then if you want to change it later, you may forget where else that constant is used. Referring to a single variable stored in a Constants struct allows you to scale these constants to use everywhere and as many times as you want, and also allows you to name the constants you're using, which also improves readability.